### ISSUE TITLE
Missing `int64_t` declaration in rectangular_lsap.cpp

### ISSUE THREAD

**Issue by gmrukwa** [scipy/scipy#11319]
I was trying to install scipy for several hours in `python:3.7-alpine` image. Unfortunately it doesn't build for me. Thought it was a problem with dependencies, tried multiple configurations, but then checked the code that's actually failing and found an issue.

#### Steps to reproduce:

Create `dummy.dockerfile` with following content:

```
FROM python:3.7-alpine
RUN apk add --no-cache build-base openblas-dev lapack-dev
RUN pip install scipy==1.4.1
```

and run:

```
docker build -t dummy -f dummy.dockerfile .
```

#### Error message:

```
creating build/temp.linux-x86_64-3.7/scipy/optimize/rectangular_lsap
  compile options: '-I/tmp/pip-build-env-psdpbyuc/overlay/lib/python3.7/site-packages/numpy/core/include -c'
  g++: scipy/optimize/rectangular_lsap/rectangular_lsap.cpp
  Running from scipy source directory.
  /tmp/pip-build-env-psdpbyuc/overlay/lib/python3.7/site-packages/numpy/distutils/system_info.py:716: UserWarning: Specified path /usr/include/python3.7m is invalid.
    return self.get_paths(self.section, key)
  scipy/optimize/rectangular_lsap/rectangular_lsap.cpp:117:43: error: 'int64_t' has not been declared
    117 | solve(int nr, int nc, double* input_cost, int64_t* output_col4row)
        |                                           ^~~~~~~
  scipy/optimize/rectangular_lsap/rectangular_lsap.cpp:186:41: error: 'int64_t' has not been declared
    186 |                                         int64_t* col4row)
        |                                         ^~~~~~~
  scipy/optimize/rectangular_lsap/rectangular_lsap.cpp:117:43: error: 'int64_t' has not been declared
    117 | solve(int nr, int nc, double* input_cost, int64_t* output_col4row)
        |                                           ^~~~~~~
  scipy/optimize/rectangular_lsap/rectangular_lsap.cpp:186:41: error: 'int64_t' has not been declared
    186 |                                         int64_t* col4row)
        |                                         ^~~~~~~
  error: Command "g++ -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -DTHREAD_STACK_SIZE=0x100000 -fPIC -I/tmp/pip-build-env-psdpbyuc/overlay/lib/python3.7/site-packages/numpy/core/include -c scipy/optimize/rectangular_lsap/rectangular_lsap.cpp -o build/temp.linux-x86_64-3.7/scipy/optimize/rectangular_lsap/rectangular_lsap.o -MMD -MF build/temp.linux-x86_64-3.7/scipy/optimize/rectangular_lsap/rectangular_lsap.o.d" failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for scipy
```

#### Scipy/Numpy/Python version information:

```
scipy==1.4.1
```

#### My findings:

I was digging a bit through the code and found that `int64_t` declaration is truly missing in the [`rectangular_lsap.cpp`](https://github.com/scipy/scipy/blob/master/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp) file. A line with

```
#include <cstdint>
```

should be added.

EDIT: @peterbell10 suggested that it may be better to

```
#include "rectangular_lsap.h"
```

**Comment by akuchotrani** [comment]
Does anyone know when this fix is going to be shipped? I am facing the same issue trying to build a docker image and installing scipy 

Are there any mitigation steps for now, for example, changing the version of the library

**Comment by ilayn** [comment]
The official 1.5.0 is scheduled to somewhere around June and probably backported to 1.4.x a bit earlier than that. For now, you can implement the fix #11320 manually and build it.

**Comment by akuchotrani** [comment]
@ilayn Thanks for the reply. I tried downgrading the scipy package to 1.3.1 and the docker build succeeded. I will upgrade to 1.5 version when the fix is in place.

**Comment by roldan** [comment]
Hi @akuchotrani, which version of python/alpine are you using when building your image?

**Comment by dianadujing** [comment]
@roldan I got same issue when using arm64v8/python:3.6-alpine and installing scipy 1.2.1

**Comment by roldan** [comment]
Hi @dianadujing, thanks for your answer! not sure but I think I also tried @akuchotrani answer but I finally resolved it using "python:3.8-slim" image (instead alpine) after reading: https://pythonspeed.com/articles/alpine-docker-python/

**Comment by nathanblair** [comment]
> Hi @dianadujing, thanks for your answer! not sure but I think I also tried @akuchotrani answer but I finally resolved it using "python:3.8-slim" image (instead alpine) after reading: https://pythonspeed.com/articles/alpine-docker-python/

This article is completely misleading. You can't compare the time it takes to build something from scratch to the time it takes to just install a pre-existing and distributed binary. `musl` is a perfectly valid `c` library, arguably more perfectly valid than `glibc`; and there's good reasons for not wanting to build `scipy`-dependent applications from `glibc` platforms (admittedly I'm assuming `python3.8-slim` is using something like `ubuntu` or `debian` as its base image).

I get that `scipy` doesn't want to have an official `musl` release; they have already done work to allow it to compile on `musl` `libc` which is certainly good enough, short of this one issue which I'm hoping to get around by upgrading to `1.6.2` (the latest at the time of this comment being made).
